runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingModeExecutionContext.java

```diff
@@ -194,6 +195,13 @@ public boolean workIsFailed() {
     return Optional.ofNullable(work).map(Work::isFailed).orElse(false);
   }
 
+  public byte[] getCurrentRecordOffset() {
+    if (activeReader == null) {
+      return NO_RECORD_OFFSET;
+    }
+    return activeReader.getCurrentRecordOffset();
```

Comment:
maybe we should do
if (activeReader == null || !activeReader.getSource().providesRecordOffsets()) {
  return NO_RECORD_OFFSET;
}
return activeReader.getCurrentRecordOffset();

Then we could change the reader implementation to verify the default reader impl to just throw an exception that it needs to be overridden if providesRecordOffsets is set to true.

I'm just trying to make it not possible to set providesRecordOffsets but forget to implement the necessary methods.

Reply:
Done.